x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/OpenAiService.java

```diff
@@ -153,6 +159,8 @@ public OpenAiModel parsePersistedConfig(String inferenceEntityId, TaskType taskT
         Map<String, Object> serviceSettingsMap = removeFromMapOrThrowIfNull(config, ModelConfigurations.SERVICE_SETTINGS);
         Map<String, Object> taskSettingsMap = removeFromMapOrThrowIfNull(config, ModelConfigurations.TASK_SETTINGS);
 
+        moveModelFromTaskToServiceSettings(taskSettingsMap, serviceSettingsMap);
+
         return createModelWithoutLoggingDeprecations(
```

Comment:
How about we rename this to something like `createModelFromPersistent` since we're not doing logging anymore.